flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -112,12 +117,23 @@ public String getEndpointId() {
 		return endpointId;
 	}
 
+	/**
+	 * Returns the rpc endpoint's state.
+	 *
+	 * <p>IMPORTANT: the state is not thread safe and can be used only in the main thread of the rpc endpoint.
+	 *
+	 * @return Rpc endpoint's state.
+	 */
+	public State getState() {
```

Comment:
Instead of exposing the internal state machine of the `RpcEndpoint` via `State`, I think it would be better to define a method `public final boolean isRunning()` which returns `true` if the state is `Running` and if not, then it returns `false`. This method could validate that it is only called from the main thread by calling `validateRunsInMainThread`.